client/src/modules/shopping-list/index.jsx

```diff
@@ -76,10 +77,19 @@ class ShoppingList extends Component {
     this.hideUpdateForm();
   };
 
-  checkIfArchived() {
+  checkIfArchived = () => {
     const { list } = this.props;
     return !list || (list && !list.isArchived);
-  }
+  };
+
+  checkIfAuthorized = () => {
+    const {
+      currentUser: { id: userId },
+      list
+    } = this.props;
+
+    return list && list.adminIds && list.adminIds.includes(userId);
```

Comment:
A method is doing two things - it is checking if list exists and checking adminIds properties.
Consider separating checkIfListExist to a separated method.

Reply:
I think it will be too complicated... In this additional method you will have to return true or false (depending on whether the list exists or not) You will have to extract list from props there to do this. In checkIfAuthorize method you'll have to extract list once again because it will be needed to access to adminIds... 
What about this solution:
I can pass list to this method to avoid checking if list exist the:

  checkIfAuthorized = list => {
    const {
      currentUser: { id: userId }
    } = this.props;

    return list.adminIds && list.adminIds.includes(userId);
  };
render() {
...
this.checkIfAuthorized(list)
...
}

Reply:
so stay at the solution you have now.
Passign list as argument doesnt make sense